src/controller/java/src/chip/onboardingpayload/QRCodeOnboardingPayloadParser.kt

```diff
@@ -118,5 +85,38 @@ class QRCodeOnboardingPayloadParser(private val mBase38Representation: String) {
       index.addAndGet(numberOfBitsToRead)
       return dest
     }
+
+    fun extractPayload(inString: String): String {
```

Comment:
This seems to crash for `""` or `A%B%C` with `StringIndexOutOfBoundsException`

How about replacing this whole thing with:

```kotlin
fun extractPayload(inString: String): String {
    return inString
        .split('%')
        .filter { s -> s.startsWith(kQRCodePrefix)}
        .map { s -> s.substring(kQRCodePrefix.length)}
        .firstOrNull() ?: ""       
}
```